src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -287,6 +314,33 @@ CHIP_ERROR DeviceControllerSystemState::Shutdown()
         mCASEServer = nullptr;
     }
 
+    if (mCASESessionManager != nullptr)
```

Comment:
Will mSystemState still work if one isn't using a CASESessionManager at all?  We have an application where we do not.

Can this be configurable to not spin up the CASESessionManager if it isn't needed?

Reply:
@msandstedt how does that application handle the fact that DeviceController always creates a CASESessionManager right now?

We could certainly make this configurable, but DeviceController assumes all over that there's a CASESessionManager, so if we might not actually have one we'd need to add a bunch of null-checks and decide what the behavior should be in all those places, right?

Reply:
In this particular case, I am in fact referring to a controller application that has its own controller code, but uses the sdk's factory to bootstrap the stack and give us an mSystemState.  It just goes to the fact that if you're writing a controller application, the factory is a convenient way to bootstrap the sdk, even if you choose not to actually use `chip::controller::DeviceController`.

I think you in fact answered my question below.  I think if the pool sizes are zero, you are left only with the CASESessionManager, which doesn't seem like much of a hit.  And I think one could apply modest patches to remove the penalty altogether.

In general, I understand the spirit of this PR and agree that a single CASESessionManager shared by all the controllers obviously makes sense.